Common/Statistics/StatisticsBuilder.cs

```diff
@@ -0,0 +1,304 @@
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using QuantConnect.Logging;
+
+namespace QuantConnect.Statistics
+{
+    /// <summary>
+    /// The StatisticsBuilder class creates summary and rolling statistics from trades, equity and benchmark points
+    /// </summary>
+    public class StatisticsBuilder
+    {
+        /// <summary>
+        /// The output summary, total and rolling statistics
+        /// </summary>
+        public StatisticsResults Results { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the StatisticsBuilder class
+        /// </summary>
+        /// <param name="trades">The list of closed trades</param>
+        /// <param name="profitLoss">Trade record of profits and losses</param>
+        /// <param name="pointsEquity">The list of daily equity values</param>
+        /// <param name="pointsPerformance">The list of algorithm performance values</param>
+        /// <param name="pointsBenchmark">The list of benchmark values</param>
+        /// <param name="startingCapital">The algorithm starting capital</param>
+        /// <param name="totalFees">The total fees</param>
+        /// <param name="totalTransactions">The total number of transactions</param>
+        public StatisticsBuilder(
+            List<Trade> trades, 
+            SortedDictionary<DateTime, decimal> profitLoss,
+            List<ChartPoint> pointsEquity, 
+            List<ChartPoint> pointsPerformance, 
+            List<ChartPoint> pointsBenchmark, 
+            decimal startingCapital, 
+            decimal totalFees, 
+            int totalTransactions)
+        {
+            var equity = ChartPointToDictionary(pointsEquity);
+
+            var firstDate = equity.Keys.FirstOrDefault().Date;
+            var lastDate = equity.Keys.LastOrDefault().Date;
+
+            var totalPerformance = GetAlgorithmPerformance(firstDate, lastDate, trades, profitLoss, equity, pointsPerformance, pointsBenchmark, startingCapital);
+            var rollingPerformances = GetRollingPerformances(firstDate, lastDate, trades, profitLoss, equity, pointsPerformance, pointsBenchmark, startingCapital);
+            var summary = GetSummary(totalPerformance, totalFees, totalTransactions);
+
+            Results = new StatisticsResults(totalPerformance, rollingPerformances, summary);
```

Comment:
could this be just a static method that returns the `StatisticsResults` directly? Maybe `Statistics.Generate(...)` to match the previous, but now it returns the new fancy `StatisticsResults` class. This would allow for more natural call site invocations, currently its:
```C#
var builder = new StatisticsBuilders(...);
var results = builder.Results;
```
But could be written:
```C#
var results = Statistics.Generate(...);
```

Thoughts?